### PR TITLE
Add isBoolean and isSet type guards

### PR DESCRIPTION
**Pull request by crgwbr**

### CODE REVIEW

File: tests/utils/is-set.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+describe('isSet', () => {
+
+   it('correctly classifies sets', () => {
+      expect(t.isSet(new Set([]))).to.strictlyEqual(true);
+      expect(t.isSet(new Set([ 'a', 'b', 'c' ]))).to.strictlyEqual(true);
+      expect(t.isSet(new Set([ 4 ]))).to.strictlyEqual(true);
+      expect(t.isSet(new Set([ 'a', 'b', 'c', 4 ]))).to.strictlyEqual(true);
+      // eslint-disable-next-line
```
Comment (onebytegone) [inline tests/utils/is-set.test.ts:12]: This `eslint-disable` needed? If so, can we disable the specific rule that failing?

**Review by onebytegone — CHANGES_REQUESTED** [review]
LGTM, just the one minor note

File: tests/utils/is-set.test.ts (hunk 1 above)
Comment (crgwbr) [reply tests/utils/is-set.test.ts]: Removed. Must've been necessary for the linting rules of 3 years ago, but not needed now.

**Review by onebytegone — APPROVED** [review]
Thank you!

**Pull request merged**